node/src/types/block.rs

```diff
@@ -1026,6 +1026,20 @@ pub(crate) mod json_compatibility {
         }
     }
 
+    impl From<JsonEraEnd> for EraEnd {
+        fn from(era_end: JsonEraEnd) -> Self {
+            let equivocators = era_end.equivocators;
+            let mut rewards: BTreeMap<PublicKey, u64> = BTreeMap::new();
+            for reward in era_end.rewards {
+                rewards.insert(reward.validator, reward.amount);
+            }
```

Comment:
Maybe slightly more canonical to map between two different collections like this:
```suggestion
            let rewards = era_end
                .rewards
                .into_iter()
                .map(|reward| (reward.validator, reward.amount))
                .collect();
```
No need for a `mut` variable then.

Reply:
Applied suggestion